pkgs/build-support/setup-hooks/audit-tmpdir.sh

```diff
@@ -15,25 +15,50 @@ auditTmpdir() {
 
     echo "checking for references to $TMPDIR/ in $dir..."
 
-    local i
-    find "$dir" -type f -print0 | while IFS= read -r -d $'\0' i; do
-        if [[ "$i" =~ .build-id ]]; then continue; fi
+    local tmpdir="$(mktemp -d)"
+    local elf_fifo="$tmpdir/elf"
+    local script_fifo="$tmpdir/script"
+    mkfifo "$elf_fifo" "$script_fifo"
 
-        if isELF "$i"; then
-            if { printf :; patchelf --print-rpath "$i"; } | grep -q -F ":$TMPDIR/"; then
-                echo "RPATH of binary $i contains a forbidden reference to $TMPDIR/"
+    # Classifier: identify ELF and script files
+    (
+        find "$dir" -type f -not -path '*/.build-id/*' -print0 \
+        | while IFS= read -r -d $'\0' file; do
+            if isELF "$file"; then
+                printf '%s\0' "$file" >&3
+            elif isScript "$file"; then
+                if [ -e "$(dirname "$file")/.$(basename "$file")-wrapped" ]; then
+                    printf '%s\0' "$file" >&4
+                fi
+            fi
+        done
+        exec 3>&- 4>&-
+    ) 3> "$elf_fifo" 4> "$script_fifo" &
+
+    # Handler: check RPATHs concurrently
+    (
+        xargs -0 -P "$NIX_BUILD_CORES" -n 1 sh -c '
```

Comment:
Technically we actually now spawning more processes than the user specified because we have two xargs.

Reply:
Yep even if the output contains zero files, there is an overhead of launching two xargs processes.
But overall in most cases the new implementation will be cheaper, because I removed overhead like calling out to `basename` and `dirname` for example.